src/main/kotlin/org/taktik/freehealth/middleware/service/impl/STSServiceImpl.kt

```diff
@@ -607,4 +605,63 @@ class STSServiceImpl(val keystoresMap: IMap<UUID, ByteArray>, val tokensMap: IMa
         return result;
     }
 
+    override fun mergeKeystores(newKeystore: String, oldKeystore: String, newPassword: String, oldPassword: String): ByteArray {
+        val newKeystoreBytes: ByteArray = try {
+            Base64.getDecoder().decode(newKeystore)
+        } catch (exception: IllegalArgumentException) {
+            throw ResponseStatusException(HttpStatus.BAD_REQUEST, "Error while base64 decoding new keystore")
+        }
+
+        val oldKeystoreBytes: ByteArray = try {
+            Base64.getDecoder().decode(oldKeystore)
+        } catch (exception: IllegalArgumentException) {
+            throw ResponseStatusException(HttpStatus.BAD_REQUEST, "Error while base64 decoding old keystore")
+        }
+
+        if (newKeystoreBytes.contentEquals(oldKeystoreBytes)) {
+            throw ResponseStatusException(HttpStatus.BAD_REQUEST, "Old and new keystores are equal")
+        }
+
+        val targetKeystore = KeyStore.getInstance("pkcs12")
+
+        try {
+            targetKeystore.load(newKeystoreBytes.inputStream(), newPassword.toCharArray())
+        } catch (exception: IOException) {
+            throw determineIOExceptionMessage(exception, "new keystore")
+        }
+
+        val source = KeyStore.getInstance("pkcs12")
+
+        try {
+            source.load(oldKeystoreBytes.inputStream(), oldPassword.toCharArray())
+        } catch (exception: IOException) {
+            throw determineIOExceptionMessage(exception, "old keystore")
+        }
+
+        val aliases = source.aliases()
+
+        while (aliases.hasMoreElements()) {
+            val alias = aliases.nextElement()
```

Comment:
Is this actually the best way to do a loop?

Reply:
No idea. Intellij autoconverted it this way from Java code